CogniteSdk.Types/Alpha/Simulators/SimulatorValue.cs

```diff
@@ -114,5 +136,83 @@ public override bool Equals(object obj)
             /// </summary>
             public override int GetHashCode() => Value.GetHashCode();
         }
+
+        /// <summary>
+        /// Simulator value that contains a list of string values.
+        /// </summary>
+        public sealed class StringArray : SimulatorValue
+        {
+            /// <summary>
+            /// Initialize new string list value.
+            /// </summary>
+            /// <param name="value">The value to set.</param>
+            public StringArray(IEnumerable<string> value)
+            {
+                Type = SimulatorValueType.STRING_ARRAY;
+                Value = value;
+            }
+
+            /// <summary>
+            /// The contained string array value.
+            /// </summary>
+            public IEnumerable<string> Value { get; set; }
+
+            /// <summary>
+            /// Return string representation of the string array value, i.e the string array itself.
+            /// </summary>
+            public override string ToString() => string.Join(", ", Value);
+
+            /// <summary>
+            /// Check if two string array values are equal.
+            /// </summary>
+            public override bool Equals(object obj)
+            {
+                return obj != null && Value.SequenceEqual((obj as StringArray).Value);
+            }
+
+            /// <summary>
+            /// Get hash code for the string array value.
+            /// </summary>
+            public override int GetHashCode() => Value.GetHashCode();
+        }
+
+        /// <summary>
+        /// Simulator value that contains a list of double values.
+        /// </summary>
+        public sealed class DoubleArray : SimulatorValue
+        {
+            /// <summary>
+            /// Initialize new double list value.
+            /// </summary>
+            /// <param name="value">The value to set.</param>
+            public DoubleArray(IEnumerable<double> value)
+            {
+                Type = SimulatorValueType.DOUBLE_ARRAY;
+                Value = value;
+            }
+
+            /// <summary>
+            /// The contained double array value.
+            /// </summary>
+            public IEnumerable<double> Value { get; set; }
+
+            /// <summary>
+            /// Return string representation of the double array value, i.e the double array itself.
+            /// </summary>
+            public override string ToString() => string.Join(", ", Value.Select(v => v.ToString(CultureInfo.InvariantCulture)));
+
+            /// <summary>
+            /// Check if two double array values are equal.
+            /// </summary>
+            public override bool Equals(object obj)
+            {
+                return obj != null && Value.SequenceEqual((obj as DoubleArray).Value);
```

Comment:
this could be simplified. The `is` pattern also implicitly checks for null

```

public override bool Equals(object obj)
{
    if (obj is DoubleArray doublearray)
    {
        // Use SequenceEqual to compare the contents of the two arrays
        return Value.SequenceEqual(doublearray.Value);
    }

    // Return false if obj is not a DoubleArray
    return false;
}